### PR TITLE
[IMP] partner_event: support multi-company

### PR DESCRIPTION
**Pull request by chienandalu**
- Add support for multi-company, so we create or associate the partners to the company configured in the event.

cc @Tecnativa

### CODE REVIEW

File: partner_event/models/event_registration.py [hunk 1]
```diff
@@ -34,16 +35,19 @@ def create(self, vals):
             Event = self.env['event.event']
             # Look for a partner with that email
             email = vals.get('email').replace('%', '').replace('_', '\\_')
+            event = Event.browse(vals['event_id'])
+            company_id = event.company_id and event.company_id.id
```
Comment (carlosdauden) [inline partner_event/models/event_registration.py]:
```suggestion
            company_id = event.company_id.id
```

File: partner_event/models/event_registration.py [hunk 2]
```diff
@@ -34,16 +35,19 @@ def create(self, vals):
...
             attendee_partner = Partner.search([
-                ('email', '=ilike', email)
+                ('email', '=ilike', email),
+                ('company_id', '=', company_id),
```
Comment (carlosdauden) [inline partner_event/models/event_registration.py:43]:
```suggestion
                '|',
                ('company_id', '=', company_id),
                ('company_id', '=', False),
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: partner_event/models/event_registration.py (hunk 2 above)
Comment (carlosdauden) [reply partner_event/models/event_registration.py:43]: Other better solution can be a second search with ('company_id', '=', False) after first search if it don't find results

Comment (chienandalu) [reply partner_event/models/event_registration.py:43]: Well, indeed if the company_id is False, that means that the partner is shared across companies, so it should be fine, isn't it?

**Review by carlosdauden — APPROVED** [review]

File: partner_event/models/event_registration.py [hunk 3]
```diff
@@ -34,16 +35,21 @@ def create(self, vals):
...
-            event = Event.browse(vals['event_id'])
             if attendee_partner:
                 vals['name'] = vals.setdefault('name', attendee_partner.name)
                 vals['phone'] = vals.setdefault(
                     'phone', attendee_partner.phone)
             elif event.create_partner:
                 # Create partner
+                vals['company_id'] = company_id
                 attendee_partner = Partner.sudo().create(
                     self._prepare_partner(vals))
```
Comment (pedrobaeza) [inline partner_event/models/event_registration.py:54]: This is unrelated, but isn't this going to generate warning due to invalid fields that are on `event.registration`, but not on `res.partner`?

**Review by pedrobaeza — APPROVED** [review]

File: partner_event/models/event_registration.py (hunk 2 above)
Comment (pedrobaeza) [reply partner_event/models/event_registration.py:43]: Yes, but it's for the improbable case of having 2 equal partners, one with the company set, and another without, giving priority to the specific company partner, but I think it's to fine grain a lot. What do you think?

**Review by rafaelbn — APPROVED** [review]
Tested functionally in runbot :+1:

**Pull request merged**